Sources/Sentry/SentryScope.m

```diff
@@ -456,16 +447,30 @@ - (SentryEvent *__nullable)applyToEvent:(SentryEvent *)event
         event.level = level;
     }
 
-    id<SentrySpan> span;
-    @synchronized(_spanLock) {
-        span = self.span;
+    NSMutableDictionary *newContext;
+    if (nil == event.context) {
+        newContext = [self context].mutableCopy;
+    } else {
+        newContext = [NSMutableDictionary new];
+        [newContext addEntriesFromDictionary:[self context]];
+        [newContext addEntriesFromDictionary:event.context];
     }
 
-    if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction] &&
-        [span isKindOfClass:[SentryTracer class]]) {
-        event.transaction = [(SentryTracer *)span name];
+    if (self.span != nil) {
+        id<SentrySpan> span;
+        @synchronized(_spanLock) {
+            span = self.span;
+        }
+        if (![event.type isEqualToString:SentryEnvelopeItemTypeTransaction] &&
+            [span isKindOfClass:[SentryTracer class]]) {
+            event.transaction = [(SentryTracer *)span name];
+        }
+        if (newContext == nil) {
+            newContext = [NSMutableDictionary new];
+        }
+        newContext[@"trace"] = [span serialize];
```

Comment:
Is the serialized form of the span what goes into `trace`? I thought it was just fixed set of values.
Span might include tags/logs etc if not now, in the future. So I believe it might not be the best idea to attach a complete span like this. 

I'm not even sure where is the right doc/spec for this. Maybe @rhcarvalho or @marandaneto can help here.

Reply:
yep, this issue actually already exists https://github.com/getsentry/develop/issues/244

Reply:
Java sets the `SpanContext`, so indeed it could be a bit cleaner, eg, no need for `tags`

Reply:
The trace context is just part of the transaction that is not a top-level event field.

See https://github.com/getsentry/sentry-go/blob/57123a409be55f61b1d5a6da93c176c55a399ad0/tracing.go#L313-L352.

The fields are:

```
TraceID
SpanID
ParentSpanID
Op
Description
Status
```